ipynb/balance_trees.ipynb

```diff
@@ -0,0 +1,488 @@
+    "\n",
+    "$\\ln \\frac{Red}{Green} = \\ln \\frac{1}{5} = - \\ln 5$\n",
+    "\n",
+    "<img src=\"images/Slide3.jpg\">\n",
+    "\n",
+    "Now, this scheme of balances can be generalized to any arbituary bifurcating trees.  But rather than comparing species abundances, we would be comparing mean log abundances of entire subtrees.  Below is how we would make this calculation on a tree with 4 leaves.\n",
+    "\n",
+    "<img src=\"images/Slide5.jpg\">\n",
+    "\n",
+    "The full equation on how to estimate balances is as follows\n",
+    "$$\n",
+    "b_i = \\sqrt{\\frac{l_i r_i}{l_i + r_i}} \n",
+    " \\ln \\big( \\frac{(\\prod_{x_n \\in L}{x_n})^{1/l_i}}\n",
+    " {(\\prod_{x_m \\in L}{x_m})^{1/r_i}} \\big)\n",
+    "$$\n",
+    "\n",
+    "where $b_i$ is the $i$th balance, $l_i$ is the number of leaves in the left subtree, $r_i$ is the number of leaves in the right subtree  $x_n$ are the abundances of the species in the left subtree, and $x_m$ are the abundances of species in the right subtree.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from __future__ import division\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from skbio.stats.composition import ilr\n",
+    "from skbio import TreeNode\n",
+    "from gneiss.balances import balanceplot, balance_basis\n",
+    "from gneiss.layouts import barchart_layout\n",
+    "import scipy\n",
+    "from functools import partial\n",
+    "from ete3 import Tree, TreeStyle, faces, AttrFace, BarChartFace, CircleFace\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To illustrate how to use these modules, consider the following tree"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "                    /-a\n",
+      "          /e-------|\n",
+      "         |          \\-b\n",
+      "-r-------|\n",
+      "         |          /-c\n",
+      "          \\f-------|\n",
+      "                    \\-d\n"
+     ]
+    }
+   ],
+   "source": [
+    "tree = TreeNode.read([u\"((a,b)e, (c,d)f)r;\"])\n",
+    "print(tree.ascii_art())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And consider the following abundances for the given sample"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "a    10\n",
+       "b    20\n",
+       "c    10\n",
+       "d    10\n",
+       "dtype: int64"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
```

Comment:
In this case no.  But there is another module where it would be really helpful to have pandas pretty printing